src/FieldWarning/Assets/Ingame/UI/InputManager.cs

```diff
@@ -160,12 +160,12 @@ void OnOrderHold()
 
         void OnOrderShortClick()
         {
-            Session.SelectionManager.DispatchMoveCommand();
+            Session.SelectionManager.DispatchMoveCommand(false);
         }
 
         void OnOrderLongClick()
         {
-            Session.SelectionManager.DispatchMoveCommand();
+            Session.SelectionManager.DispatchMoveCommand(true);
```

Comment:
I think the pattern of "need a twist on existing functionality -> add a switch in an existing method to avoid code duplication" often leads to "-> have one method doing four things controlled by two abstract switches whose meaning you always forget". It is a hefty price to pay for avoiding a bit of duplication.

My preferred approach is to factor out the bits that would be duplicated into their own methods. Example:
```
DispatchMoveCommand(bool usageMode) {
    headstatements..

    if (usageMode)
      mode1Statements
    else 
      mode2Statements

    tailstatements..
}
```
becomes:
```
DispatchMode1MoveCommand() {
    Head()
    mode1Statements
    Tail()
}

DispatchMode2MoveCommand() {
    Head()
    mode2Statements
    Tail()
}
```

Where `Head()` and `Tail()` hold the head/tail statements that would have been duplicated. The only hard part of this approach is picking good names for the duplicated code segments and the two new versions of the method.